### PR TITLE
prov/verbs: Fix regression for issue #5427

### PR DESCRIPTION
**Pull request by swelch**
Fix inversion of check introduced in commit d38c8869.

Signed-off-by: Steve Welch <swelch@systemfabricworks.com>

### CODE REVIEW

File: prov/verbs/src/verbs_domain.c [hunk 1]
```diff
@@ -351,11 +351,11 @@ fi_ibv_domain(struct fid_fabric *fabric, struct fi_info *info,
...
 	if (fi->nic && fi->nic->device_attr &&
 	    !strncmp(fi->nic->device_attr->vendor_id, "0x02c9", 6)) {
-		_domain->post_send = ibv_post_send;
-		_domain->poll_cq = ibv_poll_cq;
-	} else {
 		_domain->post_send = fi_ibv_post_send_track_credits;
 		_domain->poll_cq = fi_ibv_poll_cq_track_credits;
+	} else {
+		_domain->post_send = ibv_post_send;
+		_domain->poll_cq = ibv_poll_cq;
```
Comment (shefty) [inline prov/verbs/src/verbs_domain.c]: The intent was to have Mellanox NICs take the optimized path. Other NICs would use the path that adds the credit tracking.

**Comment by swelch** [comment]
@shefty - Knowing the intent of the original change, I updated the initialization of the verbs fi_nic structure so that Mellanox hardware will be correctly detected. Please take a look at the new commit.

**Comment by swelch** [comment]
@shefty - Added an assert that will catch if an incompatible code path is being taken when XRC is being used.

**Comment by jswaro** [comment]
3 fails. Should we be concerned with any of those? @shefty

**Comment by nikhilnanal** [comment]
continuous-integration/jenkins/pr-head — The build of this commit was aborted - this failure can be ignored.

**Comment by shefty** [comment]
The AWS failure is some sort of build issue or timeout.

The Intel CI failure is related to a missing libnuma on one of the CI nodes.

**Pull request merged**